x/profile/client/cli/tx.go

```diff
@@ -29,118 +29,56 @@ func GetTxCmd(_ string, cdc *codec.Codec) *cobra.Command {
 	--picture "https://profilePic.jpg"
 	--cover "https://profileCover.jpg"
-`, version.ClientName, version.ClientName),
-		Args: cobra.ExactArgs(1),
-		RunE: func(cmd *cobra.Command, args []string) error {
-			inBuf := bufio.NewReader(cmd.InOrStdin())
-			txBldr := auth.NewTxBuilderFromCLI(inBuf).WithTxEncoder(utils.GetTxEncoder(cdc))
-			cliCtx := context.NewCLIContextWithInput(inBuf).WithCodec(cdc)
-
-			picture := GetFlagValueOrNilOnDefault(flagProfilePic)
-			cover := GetFlagValueOrNilOnDefault(flagProfileCover)
-			var pictures *types.Pictures
-			if picture != nil && cover != nil {
-				pictures = types.NewPictures(picture, cover)
-			}
-			name := GetFlagValueOrNilOnDefault(flagName)
-			surname := GetFlagValueOrNilOnDefault(flagSurname)
-			bio := GetFlagValueOrNilOnDefault(flagBio)
-
-			msg := types.NewMsgCreateProfile(args[0], name, surname, bio, pictures, cliCtx.FromAddress)
-
-			return utils.GenerateOrBroadcastMsgs(cliCtx, txBldr, []sdk.Msg{msg})
-		},
-	}
-
-	cmd.Flags().String(flagName, "default", "Name of the profile")
-	cmd.Flags().String(flagSurname, "default", "Surname of the profile")
-	cmd.Flags().String(flagBio, "default", "Biography of the profile")
-	cmd.Flags().String(flagProfilePic, "default", "Profile related picture")
-	cmd.Flags().String(flagProfileCover, "default", "Profile related cover picture")
-
-	return cmd
-}
-
-// GetCmdEditProfile is the CLI command for editing an profile
-func GetCmdEditProfile(cdc *codec.Codec) *cobra.Command {
-	cmd := &cobra.Command{
-		Use:   "edit",
-		Short: "Edit an existent profile",
-		Long: fmt.Sprintf(`
-Edit an existing profile specifying the new moniker, name, surname, bio, a profile picture and cover.
-Every data is optional.
-
-All the attributes will not accept the string "default" as a value because it's reserved for internal operations.
-
-E.g (with all the other optional fields)
-%s tx profiles edit \
-    --moniker "DiCapLeo" \
-	--name "Leo" \
-	--surname "Di Cap" \
-	--bio "Hollywood actor. Proud environmentalist" \
-	--picture "https://profilePic.jpg"
-	--cover "https://profileCover.jpg"
 `, version.ClientName),
 		Args: cobra.ExactArgs(0),
 		RunE: func(cmd *cobra.Command, args []string) error {
 			inBuf := bufio.NewReader(cmd.InOrStdin())
 			txBldr := auth.NewTxBuilderFromCLI(inBuf).WithTxEncoder(utils.GetTxEncoder(cdc))
 			cliCtx := context.NewCLIContextWithInput(inBuf).WithCodec(cdc)
 
-			picture := GetFlagValueOrNilOnDefault(flagProfilePic)
-			cover := GetFlagValueOrNilOnDefault(flagProfileCover)
-			newMoniker := GetFlagValueOrNilOnDefault(flagNewMoniker)
-			name := GetFlagValueOrNilOnDefault(flagName)
-			surname := GetFlagValueOrNilOnDefault(flagSurname)
-			bio := GetFlagValueOrNilOnDefault(flagBio)
+			picture := viper.GetString(flagProfilePic)
+			cover := viper.GetString(flagProfileCover)
+			moniker := viper.GetString(flagMoniker)
+			name := viper.GetString(flagName)
+			surname := viper.GetString(flagSurname)
+			bio := viper.GetString(flagBio)
```

Comment:
I'm not sure if using get `GetString` having an empty string works as we would want to. What I was thinking is the following. 

Suppose you execute `desmoscli tx save --name="Riccardo"` then what will happen is that all the remaining fields are rest to `""` instead of `nil` (I've checked and they are two different values). So I think we should change it appropriately. 

This could be done by creating the following functions: 

```golang
// getOrNil returns a pointer to the given value if it is not empty. Otherwise it returns nil
func getOrNil(value string) *string {
  if len(strings.TrimSpace(value)) == 0 {
    return nil
  }
  return &value
}
```

And use such function as 

```golang
moniker := getOrNil(viper.GetString(flagMoniker))
name := getOrNil(viper.GetString(flagMoniker))
```